src/puppetlabs/puppetdb/http/reports.clj

```diff
@@ -2,31 +2,35 @@
   (:require [puppetlabs.puppetdb.query.paging :as paging]
             [puppetlabs.puppetdb.query-eng :refer [produce-streaming-body]]
             [net.cgrand.moustache :refer [app]]
+            [puppetlabs.puppetdb.http.query :as http-q]
+            [puppetlabs.puppetdb.http.events :as e]
             [puppetlabs.puppetdb.middleware :refer [verify-accepts-json validate-query-params
                                                     wrap-with-paging-options]]))
 
 (defn query-app
   [version]
   (app
-    [&]
+    [hash "events" &]
+    (let [projs ["new_value" "property" "file" "old_value"
+                 "containing_class" "line" "resource_type"
+                 "status" "resource_title" "timestamp"
+                 "containment_path" "message" ]]
+      (comp (e/events-app version) (partial http-q/project-query projs) (partial http-q/restrict-query-to-report hash)))
+
+    []
     {:get  (fn [{:keys [params globals paging-options] :as request}]
              (produce-streaming-body
                :reports
                version
                (params "query")
                paging-options
-               (:scf-read-db globals)))}))
-
-(defn build-reports-app
-  [query-app]
-  (app
-   []
-   (verify-accepts-json query-app)))
+               (:scf-read-db globals)
+               (str (:url-prefix globals) "/" (name version))))}))
 
 (defn reports-app
   [version]
-  (build-reports-app
-   (-> (query-app version)
-       (validate-query-params
+  (-> (query-app version)
+      (validate-query-params
```

Comment:
Could we factor out `validate-query-params, wrap-with-paging-options, and verify-accepts-json` to a single piece of middleware?